src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -395,7 +451,62 @@ private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncC
             HttpInstrumentationEventSource.Log.FailedProcessResult(ex);
         }
 
-        activity.Stop();
+        activity?.Stop();
+
+        if (HttpClientDuration.Enabled)
+        {
+            double durationMs;
+            if (activity != null)
+            {
+                durationMs = activity.Duration.TotalMilliseconds;
+            }
+            else
+            {
+                var endTimestamp = Stopwatch.GetTimestamp();
+                var durationS = (endTimestamp - startTimestamp) / (double)Stopwatch.Frequency;
+                durationMs = durationS * 1000;
```

Comment:
curious why we cannot rely on this always?

Reply:
I asked the same question over here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1335#discussion_r1310539091

Reply:
I don't disagree with you but it seems like a spec thing where we don't have a choice 😢

Reply:
Interesting. Will try to find the reason behind adding that line in spec. We can revisit this later

Reply:
> Interesting. Will try to find the reason behind adding that line in spec. We can revisit this later

There's more context in open-telemetry/semantic-conventions#69, and the issue to which it links (open-telemetry/specification#3520).